### ISSUE TITLE
Annotations timing is saved in a separate *.annotation~ file and not correctly read on project reload

### ISSUE THREAD

**Issue by timur-almaev** [hcmlab/nova#150]
Thanks for helping with the previous issues! Everything seems to be working now, however when I try to save the annotations I've made it saves the project file *.nova containing reference to the corresponding video file and the corresponding *.annotation file containing a discrete tier I've created. The actual timing of the annotation is however missing from both of these files and instead can be found at *.annotation~ file - I assume it's a bug because when I attempt to load the saved project back it gets the annotation timing wrong which suggests that they're saved or read in a wrong way.

**Comment by ThomasSmith3** [comment]
Are you running the current master or an older version?

**Comment by RC90** [comment]
I'm on the latest commit on master

**Comment by tobiasbaur** [comment]
The behaviour to have seperate .annotation and .annotation~ files is not a bug! The .annotation file acts as header, containing the scheme and further meta information, the annotation~ file contains the actual timings of the annotations. Nova always expects the header file to be loaded. 

Are you sure the annotation timing is wrong? I tried to reproduces this behaviour, but it works for me as expected. And if so does it also happen when you drag in the .annotation file directly instead of the project file?

**Comment by RC90** [comment]
Hi Tobias, thank you for your response and clarification. You're right, it was my bad. I just checked and everything works as expected. 

There is another thing however I don't quite understand. There are two markers in the app the green and the red. When no annotation is selected both of them follow mouse clicks on the timeline. However when I drag the mouse cursor over the timeline the red marker stops and only the green one moves. When this happens I don't see how to picture changes and it makes it hard to locate the beginning of the annotation because you can only click on the timeline and cannot drag through it. It also makes it hard to locate the end of the annotation because when you start dragging the cursor to create the annotation the red marker stops and the picture freezes. Is there a way to make the picture update whilst I drag the cursor over the timeline to create the annotation? Thank you.

**Comment by tobiasbaur** [comment]
I'm not sure if I can follow you on this one. 
In general the behaviour right now is:
If you click on the annotrack, both markers move to this position
if you click on the signaltrack (the one above the annotation) only the red marker moves.
If you hit "W" a new annotation is created between both markers.

The other way it to right-mouse button drag along the annotation tier to move a new annotation.

I dont get the point where the picture freezes, maybe you can explain this a bit more or maybe have some screenshots?

**Comment by RC90** [comment]
Let me give you an example. Suppose you create an annotation. You then want to adjust it by dragging the sides of it along the timeline. When you do this the video does not get updated, because the red marker does not move along. This makes it hard to create precise annotations.

**Comment by tobiasbaur** [comment]
Please make sure you check the "Follow Annotation" checkbox, next to the videocontrols. This should create the behaviour you want.

Also when you select an annotation you can use alt+right and left to move an annotation frame-wise.
With the Q and E buttons you can set both markers at the borders of an annotation.

**Comment by timur-almaev** [comment]
Thanks a lot, that works. I also wonder what the "auto scroll" and "correction mode" checkboxes are for? 

There is a couple of other issues I faced today. First I noticed that the red marker does not move all the way to the end of the timeline during video playback, it stops in my case about a second before the end eventhough the video itself plays till the very end of it with no problem. 

Another thing I noticed is that when I use the side playback control buttons, which I assume are intended to move the red marker to the beginning and the end of the video, it does not change the current video frame displayed. Also when I for example move the red marker to the end of the video using the button on the right and then attempt to create an annotation between the two markers the annotation is created between the green marker and the most recent frame displayed instead of the last frame of the video. Is this a bug or am I missing something?
